static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -134,27 +142,70 @@ function BaseDraggableTabList({
     });
   }, [state.collection, overflowTabs]);
 
+  const persistentTabs = [...state.collection].filter(
+    item => item.key !== 'temporary-tab'
+  );
+  const tempTab = [...state.collection].find(item => item.key === 'temporary-tab');
+
   return (
     <TabListOuterWrap style={outerWrapStyles}>
-      <TabListWrap
-        {...mergeProps(tabListProps, collectionProps)}
-        orientation={orientation}
-        hideBorder={hideBorder}
-        className={className}
-        ref={tabListRef}
-      >
-        {[...state.collection].map(item => (
-          <DraggableTab
-            key={item.key}
-            item={item}
-            state={state}
-            orientation={orientation}
-            overflowing={orientation === 'horizontal' && overflowTabs.includes(item.key)}
-            dropState={dropState}
-            ref={element => (tabItemsRef.current[item.key] = element)}
-          />
-        ))}
-      </TabListWrap>
+      <Reorder.Group axis="x" values={tabs} onReorder={setTabs} as="div">
+        <TabListWrap
+          {...mergeProps(tabListProps, collectionProps)}
+          tempViewSelected={state.selectedKey === state.collection.getLastKey()}
+          orientation={orientation}
+          hideBorder={hideBorder}
+          className={className}
+          ref={tabListRef}
+        >
+          {persistentTabs.map(item => (
+            <Reorder.Item
+              key={item.key}
+              value={tabs.find(tab => tab.key === item.key)}
+              style={{display: 'flex', flexDirection: 'row'}}
+            >
+              <DraggableTab
+                key={item.key}
+                item={item}
+                state={state}
+                orientation={orientation}
+                overflowing={
+                  orientation === 'horizontal' && overflowTabs.includes(item.key)
+                }
+                dropState={dropState}
+                onDelete={() => {
+                  setTabs(tabs.filter(tab => tab.key !== item.key));
+                }}
+                ref={element => (tabItemsRef.current[item.key] = element)}
+                isChanged
+              />
+              {state.selectedKey !== item.key &&
+                state.collection.getKeyAfter(item.key) !== state.selectedKey && (
+                  <TabDivider />
+                )}
+            </Reorder.Item>
+          ))}
+          <AddViewButton borderless size="zero" onClick={() => setIsTempTabVisible(true)}>
+            <IconAdd size="xs" style={{margin: '2 4 2 2'}} />
+            Add View
+          </AddViewButton>
+          {isTempTabVisible && <TabDivider />}
+          {isTempTabVisible && tempTab && (
```

Comment:
I think it makes more sense to render the "Add view" button and temp tab _outside_ of this component. They are separate from the list tabs and it will be easier to make this component more generic that way.

I also think if you are able to remove the implementation-specific details here, we should be ready to merge this component into master.